### ISSUE TITLE
Whitening fails for specific dataset durations

### ISSUE THREAD

**Issue by ondrzel** [gwastro/ml-mock-data-challenge-1#7]
Running the PyTorch example script over a dataset generated by `generate_data.py` with a duration of 600 seconds yields the following traceback:
```
Traceback (most recent call last):
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/data/ondzel/repos/ml-mock-data-challenge-1/examples/example_torch.py", line 624, in <module>
    main()
  File "/data/ondzel/repos/ml-mock-data-challenge-1/examples/example_torch.py", line 609, in main
    verbose=args.verbose)
  File "/data/ondzel/repos/ml-mock-data-challenge-1/examples/example_torch.py", line 533, in get_triggers
    for slice_batch, slice_times in iterable:
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 521, in __next__
    data = self._next_data()
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 561, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 49, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 49, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/data/ondzel/repos/ml-mock-data-challenge-1/examples/example_torch.py", line 323, in __getitem__
    next_slice, next_time = Slicer.__getitem__(self, index)
  File "/data/ondzel/repos/ml-mock-data-challenge-1/examples/example_torch.py", line 116, in __getitem__
    dat, t = self.generate_data(key, idxs)
  File "/data/ondzel/repos/ml-mock-data-challenge-1/examples/example_torch.py", line 100, in generate_data
    ts = ts.whiten(0.5, 0.25, low_frequency_cutoff=18.)
  File "/data/ondzel/miniconda3/envs/mlgwsc/src/pycbc/pycbc/types/timeseries.py", line 622, in whiten
    white = (self.to_frequencyseries() / psd**0.5).to_timeseries()
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/data/ondzel/miniconda3/envs/mlgwsc/src/pycbc/pycbc/types/array.py", line 66, in _convert
    return fn(self, *args)
  File "/data/ondzel/miniconda3/envs/mlgwsc/lib/python3.7/site-packages/decorator.py", line 232, in fun
    return caller(func, *(extras + args), **kw)
  File "/data/ondzel/miniconda3/envs/mlgwsc/src/pycbc/pycbc/types/array.py", line 265, in _checkother
    check_same_len_precision(self, other)
  File "/data/ondzel/miniconda3/envs/mlgwsc/src/pycbc/pycbc/types/array.py", line 128, in check_same_len_precision
    raise ValueError(msg)
ValueError: lengths do not match (1281 vs 1280)
```
Some other dataset durations, namely 660, 86400 and 2592000 seconds, do not share this issue.

**Comment by MarlinSchaefer** [comment]
This issue seems to be caused by the incorrect sample rate in #9. We should be able to fix this by hard-coding the sample rate in the examples.

**Comment by MarlinSchaefer** [comment]
Fixed by #10
